akka-docs/rst/java/code/docs/http/javadsl/server/directives/BasicDirectivesExamplesTest.java

```diff
@@ -0,0 +1,788 @@
+            .runWith(Sink.head(), mat), str -> complete(str)
+        )
+      )
+    ); // default materializer will be used
+
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("Materialized by " + materializer().hashCode()+ "!");
+    //#extractMaterializer
+  }
+
+  @Test
+  public void testWithExecutionContext() {
+    //#withExecutionContext
+
+    final ExecutionContextExecutor special =
+      ExecutionContexts.fromExecutor(Executors.newFixedThreadPool(1));
+
+    final Route sample = path("sample", () ->
+      extractExecutionContext(executor ->
+        onSuccess(() ->
+          CompletableFuture.supplyAsync(() ->
+            "Run on " + executor.hashCode() + "!", executor
+          ), str -> complete(str)
+        )
+      )
+    );
+
+    final Route route = route(
+      pathPrefix("special", () ->
+        withExecutionContext(special, () -> sample) // `special` execution context will be used
+      ),
+      sample // default execution context will be used
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("Run on " + system().dispatcher().hashCode() + "!");
+    testRoute(route).run(HttpRequest.GET("/special/sample"))
+      .assertEntity("Run on " + special.hashCode() + "!");
+    //#withExecutionContext
+  }
+
+  @Test
+  public void testExtractExecutionContext() {
+    //#extractExecutionContext
+    final Route route = path("sample", () ->
+      extractExecutionContext(executor ->
+        onSuccess(() ->
+          CompletableFuture.supplyAsync(() ->
+            "Run on " + executor.hashCode() + "!", executor // uses the `executor` ExecutionContext
+          ), str -> complete(str)
+        )
+      )
+    );
+
+    //tests:
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("Run on " + system().dispatcher().hashCode() + "!");
+    //#extractExecutionContext
+  }
+
+  @Test
+  public void testWithLog() {
+    //#withLog
+    final LoggingAdapter special = Logging.getLogger(system(), "SpecialRoutes");
+
+    final Route sample = path("sample", () ->
+      extractLog(log -> {
+        final String msg = "Logging using " + log + "!";
+        log.debug(msg);
+        return complete(msg);
+      }
+      )
+    );
+
+    final Route route = route(
+      pathPrefix("special", () ->
+        withLog(special, () -> sample)
+      ),
+      sample
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("Logging using " + system().log() + "!");
+    testRoute(route).run(HttpRequest.GET("/special/sample"))
+      .assertEntity("Logging using " + special + "!");
+    //#withLog
+  }
+
+  @Ignore("Ignore compile-only test")
+  @Test
+  public void testWithSettings() {
+    //#withSettings
+    final RoutingSettings special =(RoutingSettings) RoutingSettings.create(system().settings().config()).withFileIODispatcher("special-io-dispatcher");
```

Comment:
Cast should not be needed, `RoutingSettings.create` returns `RoutingSettings`. Also, please break up in multiple lines, the docs does not allow for very wide lines before you get a scroll bar, not sure exactly how many chars.